controllers/usersController.js

```diff
@@ -0,0 +1,142 @@
+import createError from 'http-errors';
+import jwt from 'jsonwebtoken';
+import models from '../models';
+import { env, generateToken } from '../helpers/utils';
+import Mail from '../helpers/sendMail';
+import Response from '../helpers/responseHelper';
+import { STATUS, MESSAGE } from '../helpers/constants';
+import logger from '../helpers/logger';
+
+const { User } = models;
+
+/**
+ * Class representing users controller
+ *
+ * @class UsersController
+ */
+class UsersController {
+  /**
+   * Creates a new user resource
+   *
+   * @static
+   * @param {Request} request - Request object
+   * @param {Response} response - Response object
+   * @param {Function} next - Express next function
+   * @returns {void}
+   *
+   * @memberof UsersController
+   */
+  static async register(request, response, next) {
+    const { body } = request;
+
+    try {
+      const user = await User.create(body);
+      const token = await generateToken({ user });
+      // generate confirm token
+      const confirmToken = await generateToken({ email: user.email });
+      // generate confirm link
+      const confrimLink = `${env('API_DOMAIN')}api/v1/users/confirm_account?token=${confirmToken}`;
+      // send the user a mail
+      const data = {
+        email: user.email,
+        subject: 'Account confirmation',
+        mailContext: {
+          link: confrimLink
+        },
+        template: 'signup'
+      };
+      await Mail.sendMail(data);
+      return Response.send(response, STATUS.CREATED, { token, id: user.id });
+    } catch (error) {
+      logger.error(error);
+      return next(error);
+    }
+  }
+
+  /**
+ * Get a user where the column(param) equals the specified value
+ *
+ * @static
+ * @param {string} param The column to search against (e.g email, id, username)
+ * @param {*} value The actual value to test for
+ * @returns {object} The user object
+ * @memberof UsersController
+ */
+  static async getUser(param, value) {
+    let user;
+    try {
+      user = await User.findOne({ where: { [param]: value } });
+    } catch (error) {
+      logger.log(error);
+    }
+    return user;
+  }
+
+  /**
+   * Authenticate a user
+   *
+   * @static
+   * @param {Request} request - Request object
+   * @param {Response} response - Response object
+   * @param {Function} next - Express next function
+   * @returns {void}
+   *
+   * @memberof UsersController
+   */
+  static async login(request, response, next) {
+    const { body: { email, password } } = request;
+
+    try {
+      const user = await User.findOne({ where: { email } });
+
+      // validate user password
+      if (user && !User.comparePassword(user, password)) {
+        throw createError(401, 'Invalid credentials');
+      }
+
+      // generate token from user payload
+      const payload = JSON.stringify(user.dataValues);
+      const token = await generateToken(payload);
+
+      // respond with token
+      return response
```

Comment:
**Nitpick**

- Use the `Response` helper function or follow the same output format to send responses.
This will ensure our responses are always in the same format.

Reply:
Oh, thought you must have implemented that. Nonetheless, I will make quick adjustments.

Reply:
Handle response using provided `responseHandler` helper function.